src/accountsdb/snapshots.zig

```diff
@@ -2618,12 +2618,17 @@ pub fn parallelUnpackZstdTarBall(
     // TODO: improve `zstd.Reader` to be capable of sourcing a stream of bytes
     // rather than a fixed slice of bytes, so we don't have to load the entire
     // snapshot file into memory.
-    const file_data = try allocator.alloc(u8, file_size);
-    defer allocator.free(file_data);
-    if (try file.readAll(file_data) != file_size) {
-        return error.UnexpectedEOF; // has the file shrunk since we got its size?
-    }
-    var tar_stream = try zstd.Reader.init(file_data);
+    const memory = try std.posix.mmap(
+        null,
+        file_size,
+        std.posix.PROT.READ,
+        std.posix.MAP{ .TYPE = .PRIVATE },
+        file.handle,
+        0,
+    );
+    defer std.posix.munmap(memory);
+
```

Comment:
```suggestion
try std.posix.madvise(memory.ptr, memory.len, std.posix.MADV.SEQUENTIAL | std.posix.MADV.WILLNEED);

```

Reply:
Is this assumption correct? Will it read the entire thing sequentially?

Reply:
zstd supports streaming decompression so seems likely? But can't comment beyond that

Reply:
I'm thinking this is an easy optimization we can throw in for now. Later we can redo the whole thing to stream data from the file. At least this gives us a minor improvement over normal mmap. Do you have any concerns about adding this line of code?

Reply:
No concerns, it's likely a small improvement